tests/check_ncb64.c

```diff
@@ -106,6 +106,36 @@ START_TEST(check_nc_b64_enc_n_file_foobar)
 }
 END_TEST
 
+START_TEST(check_nc_b64_enc_n_file_long_text)
+{
+        char out[8000];
+        char result[8000];
+        unsigned long len = 0;
+        FILE *fh = NULL;
+        size_t n = 8000 - 1;
+        const char *filename = TOPSRCDIR "/tests/nc_b64enc_test_files/long_text";
+        const char *resultfile = TOPSRCDIR "/tests/nc_b64enc_test_files/long_text.b64";
+
+        fh = fopen(resultfile, "r");
+
+        fseek(fh, 0, SEEK_END);
```

Comment:
I find the logic of this part of the test very strange. Open the file, seek to the end, get the position to see the length, rewind, read in that number of bytes, abort if you don't get that number of bytes. There is no check that you are not going to overflow the buffer. Doing 
      len = fread(fh, 1, sizeof(result), result);
seems much more reasonable.

It is also not clear to me why you are putting this data in external files, rather than as constants. Are you concerned about maximum constant string lengths (4096 for C99)? Your files are only about half this?

Reply:
I could have the testing collateral (long_text.b64) content in a constant, it just looks like a long string so I thought it looks better if I have it in a file instead.
The other file "long_text" has to be a file since that's what the function does converts a file into b64 string.